src/ForwardedHeader.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Primitives;
+
+namespace AspNetCore.ForwardedHttp
+{
+    //TODO: Need more efficient implementation
+    internal class ForwardedHeader
+    {
+        public ForwardedHeader(StringValues value)
+        {
+            for (var i = 0; i < value.Count; i++)
+            {
+                var header = value[i];
+                var sections = header.Split(";");
```

Comment:
I think Split breaks for the quoted string sections. E.g. `by="some;data"`

Reply:
yeah, this implementation needs work. It's just good enough for tests to be passed.
Is there any class in ASP.NET for parsing this value? The logic looks the same as query parsing.

Reply:
Cookie, Query, and Form all use very similar parsers, but be careful, they have slight differences in how they handle special characters. I don't think you'll be able to use any of them without modification.